tree/dataframe/src/RDFActionHelpers.cxx

```diff
@@ -163,6 +163,60 @@ template void MeanHelper::Exec(unsigned int, const std::vector<char> &);
 template void MeanHelper::Exec(unsigned int, const std::vector<int> &);
 template void MeanHelper::Exec(unsigned int, const std::vector<unsigned int> &);
 
+StdDevHelper::StdDevHelper(const std::shared_ptr<double> &meanVPtr, const unsigned int nSlots)
+   : fNSlots(nSlots), fResultStdDev(meanVPtr), fDistancesfromMean(nSlots, 0), fMeans(nSlots, 0), fCounts(nSlots, 0)
+{
+}
+
+void StdDevHelper::Exec(unsigned int slot, double v)
+{
+   // Online algorithm
+   auto count = ++fCounts[slot];
+   auto delta = v - fMeans[slot];
+   auto mean = fMeans[slot] + delta / count;
+   auto delta2 = v - mean;
+   auto distance = fDistancesfromMean[slot] + delta * delta2;
+
+   fCounts[slot] = count;
+   fMeans[slot] = mean;
+   fDistancesfromMean[slot] = distance;
+}
+
+void StdDevHelper::Finalize()
+{
+   double totalElements = 0;
+   for (auto c : fCounts) {
+      totalElements += c;
+   }
+   if (totalElements == 0) {
+      *fResultStdDev = 0;
+      return;
+   }
+
+   double overallMean = 0;
+   for (unsigned int i = 0; i < fNSlots; ++i) {
+      overallMean += fCounts[i] * fMeans[i];
+   }
+   overallMean = overallMean / totalElements;
+
+   double variance = 0;
+   for (unsigned int i = 0; i < fNSlots; ++i) {
+      if (fCounts[i] == 0) {
+         continue;
+      }
+      variance += fCounts[i] * (fDistancesfromMean[i] / fCounts[i] + std::pow((fMeans[i] - overallMean), 2));
+   }
+
+   variance = variance / totalElements;
```

Comment:
You put here the biased estimation of the standard deviation. Do you want the unbiased one? -> `std = sqrt( 1 / (N - 1) * ... )`

https://en.wikipedia.org/wiki/Standard_deviation

https://en.wikipedia.org/wiki/Unbiased_estimation_of_standard_deviation

Reply:
Does matter only for very small number of samples.

Reply:
`RMS` is not a proper name for this function....

```
////////////////////////////////////////////////////////////////////////////////
/// Return the Standard Deviation of an array defined by the iterators.
/// Note that this function returns the sigma(standard deviation) and
/// not the root mean square of the array.
///
/// Use the two pass algorithm, which is slower (! a factor of 2) but much more
/// precise.  Since we have a vector the 2 pass algorithm is still faster than the
/// Welford algorithm. (See also ROOT-5545)
template <typename Iterator>
Double_t TMath::RMS(Iterator first, Iterator last)
{

   Double_t n = 0;

   Double_t tot = 0;
   Double_t mean = TMath::Mean(first,last);
   while ( first != last ) {
      Double_t x = Double_t(*first);
      tot += (x - mean)*(x - mean);
      ++first;
      ++n;
   }
   Double_t rms = (n > 1) ? TMath::Sqrt(tot/(n-1)) : 0.0;
   return rms;
}
```

Reply:
Thanks, I changed the formula to the unbiased one 4d22b9610974e5856d73d856626a6e04a1095008